eZ/Bundle/EzPublishCoreBundle/Fragment/DirectFragmentRenderer.php

```diff
@@ -86,4 +97,38 @@ protected function getArguments(callable $controller, ControllerEvent $event): a
 
         return $arguments;
     }
+
+    /**
+     * @param string|\Symfony\Component\HttpKernel\Controller\ControllerReference $uri
+     *
+     * @throws \eZ\Publish\Core\MVC\Symfony\Templating\Exception\InvalidResponseException
+     */
+    public function render(
+        $uri,
+        Request $request,
+        array $options = []
+    ): Response {
+        $event = $this->getControllerEvent($request);
+        $controller = $this->getController($event);
+        $arguments = $this->getArguments($controller, $event);
+
+        $response = call_user_func_array($controller, $arguments);
+
+        if ($response instanceof Response) {
+            return $response;
+        } elseif ($response instanceof View) {
+            return new Response($this->viewTemplateRenderer->render($response));
+        } elseif (is_string($response)) {
+            return new Response($response);
+        }
+
+        throw new InvalidResponseException(
+            sprintf('Unsupported type (%s)', is_object($response) ? get_class($response) : gettype($response))
```

Comment:
Just would like to add that in PHP 8.0 (and in Symfony polyfill for PHP8: https://github.com/symfony/polyfill-php80) a `get_debug_type` function exists that does this.

Reply:
All Symfony polyfills are marked as `internal`. Are you sure it is safe to use them?

Reply:
I'm advocating for their use since functions they provide are part of new PHP versions. Cost of their introduction is low (since they are not loaded into memory if they're not used, and if functions already exist - because we're already on new PHP version - they are not loaded at all).

I don't know the reasoning for them being marked internal, but clearly Symfony intends for them to be used.

However, I do understand that adding them here would mean introducing a dependency, even if small, to all projects using ezplatform-kernel.